boa/vyper/contract.py

```diff
@@ -702,90 +683,32 @@ def unoptimized_bytecode(self):
         )
         return s + self.data_section
 
-    # compile a fragment (single expr or statement) in the context
-    # of the contract, returning the ABI encoded value if it is an expr.
-    def compile_stmt(self, source_code: str) -> Any:
-        # this method is super slow so we cache compilation results
-        if source_code in self._eval_cache:
-            return self._eval_cache[source_code]
-
-        ast = parse_to_ast(source_code)
-        vy_ast.folding.fold(ast)
-        ast = ast.body[0]
-
-        typ = None
-        return_sig = ""
-        debug_body = source_code
-
-        ifaces = self.compiler_data.interface_codes
-
-        if isinstance(ast, vy_ast.Expr):
-            with self.override_vyper_namespace():
-                try:
-                    ast_typ = get_exact_type_from_node(ast.value)
-                    return_sig = f"-> {ast_typ}"
-                    debug_body = f"return {source_code}"
-                except InvalidType:
-                    pass
-
-        # wrap code in function so that we can easily generate code for it
-        wrapper_code = textwrap.dedent(
-            f"""
-            @external
-            @payable
-            def __boa_debug__() {return_sig}:
-                {debug_body}
-        """
-        )
-
-        ast = parse_to_ast(wrapper_code, ifaces)
-
-        # splice in constant definitions so that eval("MY_CONSTANT") works
-        for constant_def in self.compiler_data.vyper_module.get_children(
-            vy_ast.VariableDecl, {"is_constant": True}
-        ):
-            ast.add_to_body(constant_def)
-        vy_ast.folding.fold(ast)
-        ast.body = [ast.body[0]]
-
-        # annotate ast
-        with self.override_vyper_namespace():
-            validation.add_module_namespace(ast, self.compiler_data.interface_codes)
-            validation.validate_functions(ast)
+    def eval(self, stmt: str, value: int = 0, gas: int = None) -> Any:
+        """eval vyper code in the context of this contract"""
+        tmp = self._source_map
 
-        ast = ast.body[0]
+        # this method is super slow so we cache compilation results
+        if stmt in self._eval_cache:
```

Comment:
not thrilled about moving the caching code here but hmm

Reply:
I think we can entirely remove the cacheing part since well: do we even need it now? i think cacheing makes sense with internal fns but random stmts, meeeeeeh

Reply:
i think caching is important because it will end up being a hotspot in user code somewhere